presto-router/src/test/java/com/facebook/presto/router/TestRouterAuthentication.java

```diff
@@ -0,0 +1,253 @@
+        runKeytoolCommand("%s/bin/keytool -genkeypair -alias localhost -keyalg RSA -keysize 2048 -validity 365 -keystore keystore.jks -storepass %s -keypass %s -dname CN=localhost -ext SAN=ip:127.0.0.1", javaHome, jksPassword, jksPassword);
+        runKeytoolCommand("%s/bin/keytool -exportcert -alias localhost -keystore keystore.jks -file certificate.cer -storepass %s", javaHome, jksPassword);
+        runKeytoolCommand("%s/bin/keytool -importcert -alias localhost -file certificate.cer -keystore truststore.jks -storepass %s -noprompt", javaHome, jksPassword);
+        keystorePath = storePath.resolve("keystore.jks");
+        truststorePath = storePath.resolve("truststore.jks");
+    }
+
+    @BeforeClass
+    public void setup()
+            throws Exception
+    {
+        Logging.initialize();
+        Path tempFile = Files.createTempFile("temp-config", ".json");
+        configFile = getConfigFile(ImmutableList.of(createPrestoServer()), tempFile.toFile());
+
+        authenticatorPropertiesFile = Paths.get("etc/password-authenticator.properties");
+        authenticatorPropertiesFile.getParent().toFile().mkdirs();
+
+        Path passwordFilePath = Files.createTempFile("passwords", ".db");
+        // credentials come from htpasswd -n -B -C 8, input "testpass"
+        Files.write(passwordFilePath, "testuser:$2y$08$KBfSimK6KTZFyCKlJACpTu7VMBHlnFixXm8tDh9I0rDf3IIuobtHy".getBytes(UTF_8));
+        Files.write(
+                authenticatorPropertiesFile,
+                format("password-authenticator.name=file\nfile.password-file=%s", passwordFilePath).getBytes(UTF_8),
+                StandardOpenOption.CREATE);
+        storePath = Files.createTempDirectory("jks-store");
+        setupCertStores();
+
+        Bootstrap app = new Bootstrap(
+                new TestingNodeModule("test"),
+                new TestingHttpServerModule(),
+                new JsonModule(),
+                new JaxrsModule(true),
+                new RouterSecurityModule(),
+                new RouterModule());
+
+        Injector injector = app.doNotInitializeLogging()
+                .setRequiredConfigurationProperty("router.config-file", configFile.getAbsolutePath())
+                .setRequiredConfigurationProperty("presto.version", "test")
+                .setOptionalConfigurationProperty("http-server.authentication.type", "PASSWORD")
+                .setOptionalConfigurationProperty("http-server.http.enabled", "false")
+                .setOptionalConfigurationProperty("http-server.https.enabled", "true")
+                .setOptionalConfigurationProperty("http-server.https.keystore.path", keystorePath.toAbsolutePath().toString())
+                .setOptionalConfigurationProperty("http-server.https.keystore.key", jksPassword)
+                .initialize();
+        injector.getInstance(RouterPluginManager.class).loadPlugins();
+
+        // Add auth on the fly to avoid needing to write etc/password-authenticator.properties
```

Comment:
I have a question regarding this comment, aren't we writing password-authenticator.properties above?